addon/components/stacked-vertical-bar-chart.js

```diff
@@ -0,0 +1,1015 @@
+    var originalOrder = this.get('originalBarOrder');
+    return (barData1, barData2) => {
+      return this.defaultCompareFn(originalOrder.indexOf(barData1.barLabel),
+                                   originalOrder.indexOf(barData2.barLabel));
+    };
+  }),
+
+  /**
+   * The current bar sorting function, depending on what barSortKey is.
+   * @type {function}
+   */
+  barSortingFn: Ember.computed('valueBarSortingFn', 'customBarSortingFn', 'originalBarSortingFn', 'barSortKey', function() {
+    var barSortKey = this.get('barSortKey');
+    if (barSortKey === 'value') {
+      return this.get('valueBarSortingFn');
+    } else if (barSortKey === 'custom') {
+      return this.get('customBarSortingFn');
+    } else if (barSortKey === 'original' || Ember.isNone(barSortKey)) {
+      return this.get('originalBarSortingFn');
+    } else {
+      throw new Error("Invalid barSortKey");
+    }
+  }),
+
+  /**
+   * Array containing an object for each bar. These objects contain the barLabel
+   * and net value for each bar. Used for bar sorting in `barNames`.
+   * @type {Array.<Object>}
+   */
+  netBarValues: Ember.computed('dataGroupedByBar', function() {
+    var dataGroupedByBar = this.get('dataGroupedByBar');
+    return _.map(dataGroupedByBar, (barData, barLabel) => {
+      var barValue = barData.reduce((sum, slice) => {
+        return sum + slice.value;
+      }, 0);
+      return { barLabel: barLabel, value: barValue };
+    });
+  }),
+
+  /**
+   * Order in which bars should appear in the chart, by bar label. This list
+   * is sorted using the appropriate barSortingFn.
+   * @see barSortingFn
+   * @type {Array.<string>}
+   */
+  barNames: Ember.computed('netBarValues', 'barSortingFn', 'barSortAscending', function() {
+    var sortedBars, sortedBarNames;
+    sortedBars = this.get('netBarValues').sort(this.get('barSortingFn'));
+    sortedBarNames = _.pluck(sortedBars, 'barLabel');
+    if (!this.get('barSortAscending')) {
+      sortedBarNames.reverse();
+    }
+    return sortedBarNames;
+  }),
+
+  /**
+   * Explicitly written version of the default comparison function that is used
+   * by Array#sort. Used by every slice and bar comparison functions that are
+   * comparing specific parameters.
+   * @function
+   */
+  defaultCompareFn: function(reference1, reference2) {
+    if (reference1 < reference2) {
+      return -1;
+    } else if (reference1 > reference2) {
+      return 1;
+    } else {
+      return 0;
+    }
+  },
+
+  // ---------------------------------------------------------------------------
+  // Layout
+  // ---------------------------------------------------------------------------
+
+  labelHeightOffset: Ember.computed('_shouldRotateLabels', 'maxLabelHeight', 'labelHeight', 'labelPadding', function() {
+    var labelSize;
+
+    if (this.get('_shouldRotateLabels')) {
+      labelSize = this.get('maxLabelHeight');
+    } else {
+      labelSize = this.get('labelHeight');
```

Comment:
Added `labelHeight` to the public API documentation. Also added a comment here saying where `labelHeight` comes from